File: order-service/src/main/java/com/intexsoft/webshop/orderservice/dto/order/OrderCreateDto.java
package com.intexsoft.webshop.orderservice.dto.order;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.intexsoft.webshop.orderservice.dto.detail.DetailCreateDto;
import com.intexsoft.webshop.orderservice.model.enums.PaymentMethod;
import com.intexsoft.webshop.orderservice.validation.ValueOfEnum;
import jakarta.annotation.Nullable;
import jakarta.validation.constraints.*;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@JsonIgnoreProperties(ignoreUnknown = true)
public class OrderCreateDto {
    @NotNull
    @Positive
    Long pickupPointId;
    @NotNull
    @Positive
    Long shopId;
    @NotNull
    @Positive
    Long userId;
    @NotBlank
    @ValueOfEnum(enumClass = PaymentMethod.class)
    String paymentMethod;
    @Nullable
    @Pattern(regexp = "^(?!\\s*$).+")
    String comment;
    @NotEmpty
    @JsonProperty("orderDetails")
    List<DetailCreateDto> detailCreateDtos;
}

File: order-service-kt/src/main/kotlin/com/intexsoft/webshop/orderservicekt/dto/order/OrderCreateDto.kt
package com.intexsoft.webshop.orderservicekt.dto.order

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import com.intexsoft.webshop.orderservicekt.dto.detail.DetailCreateDto
import com.intexsoft.webshop.orderservicekt.model.enums.PaymentMethod
import jakarta.validation.Valid
import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.Positive
@JsonIgnoreProperties(ignoreUnknown = true)
data class OrderCreateDto(
    @field:Positive
    val pickupPointId: Long,
    @field:Positive
    val shopId: Long,
    @field:Positive
    val userId: Long,
    val paymentMethod: PaymentMethod,
    val comment: String?,
    @JsonProperty("orderDetails")
    @field:NotEmpty
    @field:Valid
    val detailCreateDtos: List<DetailCreateDto>
)